internal/output.go

```diff
@@ -63,172 +65,160 @@ func processOutput(params *processOutputParams) {
-			failedTests = append(failedTests, event.Test)
-		}
-
+		// Print Package PASS / FAIL lines
+		var outLine string
 		if event.Test == "" && (event.Action == "pass" || event.Action == "fail") {
-			if event.Action == "pass" && dummyPackages[event.Package] {
-				printSkipped(event.Package)
-				continue
-			}
+
 			if event.Action == "pass" {
 				outLine = shColor("green", "✔ ") + shColor("reset:bold", event.Package)
-			} else {
-				outLine = shColor("red", "◼ ") + shColor("reset:bold", event.Package)
+			}
+
+			if event.Action == "fail" {
 				pkgsFailed = append(pkgsFailed, event.Package)
+				outLine = shColor("red", "◼ ") + shColor("reset:bold", event.Package)
 			}
+
 			outLine += strings.Repeat(" ", maxPkgLen-len(event.Package))
 
-			pastCoverage := prevCoverages[event.Package]
-			delete(prevCoverages, pastCoverage)
-			if regexCoverageNoStatements.MatchString(pastCoverage) {
-				outLine += "   "
-				outLine += shColor("gray", sf("%6s", "-")+"     no statements")
+			// Build package coverage + elapsed
+			prevCoverage := prevCoverages[event.Package]
+			if regexCoverageNoStatements.MatchString(prevCoverage) {
+				outLine += "   " + shColor("gray", sf("%6s", "-")+"     no statements")
+
 			} else {
-				pkgCoverage := regexCoverageNonZero.ReplaceAllString(pastCoverage, "$1")
+				pkgCoverage := regexCoverageNonZero.ReplaceAllString(prevCoverage, "$1")
 				c := coverageParse(pkgCoverage)
 				coverages = append(coverages, c)
-
-				outLine += "   "
-				outLine += shColor(coverageColor(c), sf("%6s", pkgCoverage))
-
-				outLine += "     "
+				outLine += "   " + shColor(coverageColor(c), sf("%6s", pkgCoverage)) + "     "
 
 				if event.Elapsed == 0 {
-					// slight heuristics - elipsed 0 == cached (we cannot tell otherwise from the JSON)
-					outLine += shColor("gray", "cached")
+					outLine += shColor("gray", "cached") // elapsed 0 == cached (we cannot tell otherwise from the JSON)
 				} else {
 					outLine += fmt.Sprintf("%.3fs", event.Elapsed)
 				}
 			}
 
 			if params.FlagVerbose {
+				// print a separator between packages
 				outLine += "\n" + shColor("gray", strings.Repeat("-", maxPkgLen+22))
 			}
 		}
-		lineOutTrimmed(outLine)
-	}
 
-	if params.FailedTests != nil {
-		*params.FailedTests = failedTests
+		lineOutTrimmed(outLine)
 	}
 
 	params.LineOut()
 
+	// Print summary
 	chev := shColor("gray", "❯")
-	avgCoverage := sliceAvg(coverages)
-	cover := sf("%.2f", avgCoverage) + "%"
-	if params.AverageCoverage {
-		params.LineOut(sf("%s Average Coverage: %s", chev, shColor(coverageColor(avgCoverage)+":bold", cover)))
-	}
 	pkgs := sf("tested: %d", len(params.ToTestPackages))
 	pkgs += shColor("red", sf("    failed: %d", len(pkgsFailed)))
 	pkgs += shColor("yellow", sf("    noTests: %d", len(pkgsNoTests)))
 	pkgs += shColor("gray", sf("    excluded: %d", len(params.IgnoredPackages)))
 	params.LineOut(sf("%s Pkgs: %s", chev, pkgs))
 
-	params.LineOut()
+	// Print coverage
```

Comment:
Moved below summary for consistency with where we print when fullCoverage is set